### ISSUE TITLE
Support for more custom fields 

### ISSUE THREAD

**Issue by MaksadAnnamuradov** [perry-mitchell/webdav-client#265]
I am trying to read the nextCloud directory using webdav client and I need more custom fields such as <oc:permissions/>, <oc:share-types />, <oc:fileid />. I tried to getDirectoryContents with {details: true} but it is still not giving those above 3 options that I am looking for. Is there way to add those custom fields to the response? Or is there way that I can give an customOptions with those fields and send the request? I have tried many other packages and I could not get them to give me those 3 fields and I really need them in my project. Thanks!

**Comment by perry-mitchell** [comment]
Yes, perhaps `getDirectoryContents` doesn't return a full prop listing. Can you try `stat` instead on a single file/dir to see if it returns what you need in the `props` sub property? Please refer to [this documentation](https://github.com/perry-mitchell/webdav-client#item-stats).
